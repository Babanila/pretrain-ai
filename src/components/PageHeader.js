import React from "react";
import { cx, css } from "emotion";

const pageHeaderDiv = css`
  width: 100%;
  font-family: "Proxima-nova", "sans-serif", "Source Sans Pro";
  color: #ffffff;
  background-color: #3c4d58;
  padding-left: 1em;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 100;
`;

function PageHeader() {
  return (
    <div className={cx(pageHeaderDiv)}>
      <h1>Pretrain AI Data</h1>
    </div>
  );
}

export default PageHeader;
