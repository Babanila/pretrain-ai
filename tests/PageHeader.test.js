import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import PageHeader from "../src/components/PageHeader";

describe("Check PageHeader Component", () => {
  it("should renders without crashing", () => {
    const tree = renderer.create(<PageHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correct number of element in PageHeader component", () => {
    const PageHeaderWrapper = shallow(<PageHeader />);
    const actual = PageHeaderWrapper.find("div");

    expect(PageHeader).toBeDefined();
    expect(actual.exists).toBeTruthy();
    expect(actual.length).toEqual(1);
    expect(actual.find("h1").length).toEqual(1);
    expect(actual.find("h1").text()).toEqual("Pretrain AI Data");
  });
});
