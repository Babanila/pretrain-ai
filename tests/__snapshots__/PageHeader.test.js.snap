// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check PageHeader Component should renders without crashing 1`] = `
<div
  className="css-mxp23c"
>
  <h1>
    Pretrain AI Data
  </h1>
</div>
`;
